### PR TITLE
boards: arm: Add support for the VBLUno52 board

### PR DESCRIPTION
**Pull request by iotmember**
+ The VBLUno52 board
    nRF52832 ARM Cortex-M4F processor
    Bluetooth Low Energy 5.0
    DAPLink interface
    UNO pinout
    4 power

+ The following samples were tested:
    hello_world
    basic/button
    basic/blinky
    bluetooth/peripheral_hr
    bluetooth/beacon

Signed-off-by: Nam Do <robotden@gmail.com>

### CODE REVIEW

**Comment by iotvietmember** [comment]
@dbkinder @galak  Would you please check my commit? Thanks all!

**Comment by iotvietmember** [comment]
@galak @dbkinder I hope that this commit will be checked soon. Thanks

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+     nRF52_VBLUno52 board
+
+Hardware
+********
+
+The VBLUno52 board has two external oscillators. The frequency of
+the slow clock is 32.768 kHz. The frequency of the main clock
+is 64 MHz.
+
+Supported Features
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: Just checking, while the board may support these features, are all these features supported by Zephyr?  (I'm not aware of any FOTA support in Zephyr)

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	+ 1 Led and 1 Button onboard.
+	+ GPIO Voltage: 0 – 3.3V.
+- *DAPLink (CMSIS-DAP) interface for program and debug*:
+	+ USB MSD: Drag and Drop programming flash memory.
+	+ USB HID (DAP): CMSIS-DAP compliant debug channel.
+	+ USB CDC: Virtual COM port for log, trace and terminal emulation.
+- Supports hardware flow control features (RTS/CTS).
+- *Energy monitoring for BLE module by current measurement*
+- FOTA (Firmware over the air): Upgrade firmware over BLE interface.
+- Build good applications with:
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: Simililar question.  This looks like a cut/paste of generic board information, but we should focus on what Zephyr supports (and leave our discussion about other OS support such as Arduino, RIOT-OS, etc.)

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+* BUTTON = BUT = SW0 = P0.17
+
+Programming and Debugging
+*************************
+
+Flashing
+========
+
+The VBLUno52 board has on-board DAPLink (CMSIS-DAP) interface for flashing and debugging.
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: has an on-board

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	+ NFC Tag.
+	+ 21 digital channels, 6 ADC 10bit channels.
+	+ 1 Led and 1 Button onboard.
+	+ GPIO Voltage: 0 – 3.3V.
+- *DAPLink (CMSIS-DAP) interface for program and debug*:
+	+ USB MSD: Drag and Drop programming flash memory.
+	+ USB HID (DAP): CMSIS-DAP compliant debug channel.
+	+ USB CDC: Virtual COM port for log, trace and terminal emulation.
+- Supports hardware flow control features (RTS/CTS).
+- *Energy monitoring for BLE module by current measurement*
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: Why is this in italic? (and does Zephyr support this?)

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+========
+
+The VBLUno52 board has on-board DAPLink (CMSIS-DAP) interface for flashing and debugging.
+You do not need any other programming device.
+You only need to install the pyOCD tool (https://pypi.python.org/pypi/pyOCD)
+
+This tutorial uses the blinky application :ref:`blinky-sample`
+
+See the :ref:`getting_started` for general information on setting up
+ your development environment.
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: delete the extra space before "your"

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+#. To build and flash the generated binary image to the board, enter:
+
+   .. code-block:: console
+
+      $ make BOARD=nrf52_vbluno52 flash
+
+Testing the VBLUno52 with Zephyr: buttons, LEDs, UART, BLE
+**********************************************************
+
+ Here are some sample applications that you can use to test different
```
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: Delete the leading space on these next 8 lines

**Review by dbkinder — CHANGES_REQUESTED** [review]
Some questions about board features supported by Zephyr, and some editing suggestions

**Comment by iotvietmember** [comment]
@dbkinder  I have just fixed those problems

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
++-----------+------------+----------------------+
+| RADIO     | on-chip    | Bluetooth            |
++-----------+------------+----------------------+
+| RTT       | on-chip    | console              |
++-----------+------------+----------------------+
+| I2C       | on-chip    | i2c                  |
++-----------+------------+----------------------+
+| SPI       | on-chip    | spi                  |
++-----------+------------+----------------------+
+Other hardware features are not supported by the Zephyr kernel.
```
Comment (galak) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: there should be a blank line after the table.

**Review by galak — CHANGES_REQUESTED** [review]
Need a blank line after table on line 63 of rst

File: boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst (hunk 7 above)
Comment (dbkinder) [inline boards/arm/nrf52_vbluno52/doc/nrf52_vbluno52.rst]: yep.  good catch.

**Comment by iotvietmember** [comment]
@galak @dbkinder I have just fixed the error of rst. Many thanks

**Review by dbkinder — APPROVED** [review]
+1 for docs

**Comment by galak** [comment]
merged a version #782

**Pull request closed without merging**